common/ActionStatusSummary.ts

```diff
@@ -36,27 +36,38 @@ const getCacheKey = (
 
 export const getStatusSummary = memoize(_getStatusSummary, getCacheKey);
 
-export const getStatusColor = (color: string, theme: Theme) => {
+export const getThemeColor = (color: string, theme: Theme) => {
   return theme.graphColors[color];
 };
 
-interface ActionWithStatusSummary {
-  statusSummary?: { color?: string; identifier?: string };
+export interface ActionWithStatusSummary {
+  statusSummary?: {
+    color?: string;
+    identifier?: ActionStatusSummaryIdentifier;
+    label?: string;
+  };
+  color?: string | null;
 }
 
 export const getStatusColorForAction = (
   action: ActionWithStatusSummary,
   plan: PlanContextType,
   theme: Theme
 ) => {
-  const c = action?.color;
-  if (c != null) {
-    return getStatusColor(c, theme);
+  const { color } = action;
```

Comment:
Tiny nitpick: I think it's a bit clearer if we only destructure the action object once

```suggestion
  const { color, statusSummary } = action;
```